enforcer-rules/src/main/java/com/google/cloud/tools/dependencies/enforcer/LinkageCheckerRule.java

```diff
@@ -270,4 +287,32 @@ public void execute(@Nonnull EnforcerRuleHelper helper) throws EnforcerRuleExcep
       throw new EnforcerRuleException("Failed to collect dependency " + ex.getMessage(), ex);
     }
   }
+
+  private static String findPaths(DependencyNode root, Artifact artifact) {
+    ImmutableList.Builder<ImmutableList<DependencyNode>> result = ImmutableList.builder();
+
+    Deque<DependencyNode> stack = new ArrayDeque<>();
+    stack.addLast(root);
+    findArtifact(result, root, stack, artifact);
+
+    ImmutableList<String> paths =
+        result.build().stream().map(path -> Joiner.on(" > ").join(path)).collect(toImmutableList());
+    // Joining one or more paths to the artifact
+    return Joiner.on("\n").join(paths);
+  }
+
+  private static void findArtifact(
+      ImmutableList.Builder<ImmutableList<DependencyNode>> result,
+      DependencyNode node,
+      Deque<DependencyNode> path,
+      Artifact artifact) {
+    if (Artifacts.toCoordinates(node.getArtifact()).equals(Artifacts.toCoordinates(artifact))) {
```

Comment:
Any chance node.getArtifact returns null here? I ask because we're specifically dealing with missing artifacts.

Reply:
No, `node.getArtifact()` does not return null. `node.getArtifact().getFile()` may return null if artifact is missing.